lib/good_job.rb

```diff
@@ -1,19 +1,17 @@
 require "rails"
-require 'good_job/railtie'
 
-require 'good_job/configuration'
-require 'good_job/log_subscriber'
-require 'good_job/lockable'
-require 'good_job/job'
-require 'good_job/scheduler'
-require 'good_job/multi_scheduler'
-require 'good_job/adapter'
-require 'good_job/performer'
-require 'good_job/current_execution'
-require 'good_job/notifier'
+require "active_job"
+require "active_job/queue_adapters"
+require "good_job/railtie"
```

Comment:
When not requiring the Railtie the initializers were not triggered in railtie.rb file due to loading orders. 

To verify this assumption I did two things: 

1. Use `loader.eager_load` after `loader.setup` which worked.
2. Enable eager loading in test.rb file of spec/test_app which worked

Thus moved the railtie to good_job top requires. 

I usually require railtie on top of my main gem entry class as well. Opinions appreciated

Reply:
I ran across this before in Zeitwerk being troublesome and did some [Github searching](https://github.com/search?l=&q=zeitwerk+%22railtie%22+path%3Alib&type=Code).

I think this should happen at the bottom of the file _after_ Zeitwerk is set up and _after_ the root namespace is created. Here's an example: https://github.com/dukex/aggregated_record/blob/e8a597aaee94eb6e03ba0a26654a9c7f50a9c321/lib/aggregated_record.rb#L4-L13

Reply:
Changed